defrag/modules/sync_utils.py

```diff
@@ -0,0 +1,61 @@
+# Defrag - centralized API for the openSUSE Infrastructure
+# Copyright (C) 2021 openSUSE contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+from functools import partial, wraps
+from typing import Awaitable, Callable
+
+__MOD_NAME__ = "sync_utils"
```

Comment:
I think we should not have this as a module, since removing this module would potentially break other modules in the future, which is kind of against the hole modularization thing. I think we should move this to `modules/helpers/`